File: ios/MonProfPerso/Screens/ManageScreens.swift
import SwiftUI

// MARK: - Écran 31, Gérer le cours (annuler / reprogrammer)
struct ManageCourseScreen: View {
    @EnvironmentObject var router: Router
    @State private var slot = 1
    var body: some View {
        AkScreen {
            TopBar(title: "Détail du cours", onBack: { router.back() })
            ScrollView {
                VStack(alignment: .leading, spacing: 0) {
                    VStack(alignment: .leading, spacing: 0) {
                        HStack {
                            Text("Maths · 3ᵉ").font(AkFont.schibstedExtra(16)).foregroundColor(Ak.ink)
                            Spacer()
                            HStack(spacing: 5) { Circle().fill(Ak.online).frame(width: 6, height: 6); Text("Confirmé").font(AkFont.bold(11)).foregroundColor(Ak.green) }
                                .padding(.horizontal, 10).padding(.vertical, 5).background(Ak.greenSoft).clipShape(RoundedRectangle(cornerRadius: 8))
                        }
                        HStack(spacing: 11) {
                            InitialsAvatar(initials: "KN", size: 42, radius: 12, fontSize: 15)
                            VStack(alignment: .leading, spacing: 0) { Text("Koffi N'Guessan").font(AkFont.bold(14)).foregroundColor(Ak.ink); Text("Professeur de Maths").font(AkFont.regular(12)).foregroundColor(Ak.muted) }
                            Spacer()
                            IconSquare(system: "bubble.left", bg: Ak.cardField, tint: Ak.green, size: 38) { router.go(.messaging) }
                        }.padding(.top, 13)
                        Rectangle().fill(Ak.cardField).frame(height: 1).padding(.vertical, 13)
                        iconLine("calendar", "Samedi 22 juin · 16h00 · 1h30")
                        iconLine("house.fill", "À domicile · Cocody, Riviera 3")
                        iconLine("creditcard.fill", "6 000 F · payé via Orange Money")
                    }.akCard(radius: 18, padding: 16)

                    HStack(alignment: .top, spacing: 10) {
                        Image(systemName: "info.circle.fill").font(.system(size: 17)).foregroundColor(Ak.orange)
                        VStack(alignment: .leading, spacing: 3) {
                            Text("Conditions d'annulation").font(AkFont.bold(13)).foregroundColor(Ak.ink)
                            Text("Gratuit jusqu'à 24h avant le cours. Au-delà, 50% du montant est facturé au professeur pour son déplacement.")
                                .font(AkFont.regular(12)).foregroundColor(Color(hex: 0x7A5733)).lineSpacing(3)
                        }
                    }.padding(14).background(Color(hex: 0xFBF1E6)).clipShape(RoundedRectangle(cornerRadius: 15, style: .continuous)).padding(.top, 14)

                    Text("Reprogrammer, nouveaux créneaux").font(AkFont.schibstedBold(14)).foregroundColor(Ak.ink).padding(.top, 14)
                    HStack(spacing: 8) { rChip("Dim 23 · 14h", 0); rChip("Lun 24 · 17h", 1); rChip("Mar 25 · 16h", 2); Spacer() }.padding(.top, 11)
                }.padding(.horizontal, 22).padding(.top, 16)
            }
            HStack(spacing: 10) {
                Text("Annuler").font(AkFont.bold(14)).foregroundColor(Ak.danger).frame(maxWidth: .infinity).padding(.vertical, 15)
                    .overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.danger, lineWidth: 1))
                    .contentShape(Rectangle()).onTapGesture { router.go(.myCourses) }
                Text("Confirmer le report").font(AkFont.bold(14)).foregroundColor(.white).frame(maxWidth: .infinity).padding(.vertical, 15)
                    .background(Ak.green).clipShape(RoundedRectangle(cornerRadius: 14))
                    .contentShape(Rectangle()).onTapGesture { router.go(.myCourses) }
            }.padding(.horizontal, 22).padding(.vertical, 14).background(.white)
        }
    }
    func iconLine(_ icon: String, _ text: String) -> some View {
        HStack(spacing: 9) { Image(systemName: icon).font(.system(size: 14)).foregroundColor(Ak.green); Text(text).font(AkFont.regular(13)).foregroundColor(Ak.textBody) }.padding(.bottom, 8)
    }
    func rChip(_ t: String, _ idx: Int) -> some View {
        let sel = slot == idx
        return Text(t).font(sel ? AkFont.bold(12.5) : AkFont.semibold(12.5)).foregroundColor(sel ? .white : Ak.inkSoft)
            .padding(.horizontal, 14).padding(.vertical, 9).background(sel ? Ak.green : .white).clipShape(RoundedRectangle(cornerRadius: 11, style: .continuous))
            .overlay(RoundedRectangle(cornerRadius: 11).stroke(Ak.border, lineWidth: sel ? 0 : 1)).onTapGesture { slot = idx }
    }
}

// MARK: - Écran 32, Bilan de progression détaillé
struct ProgressReportScreen: View {
    @EnvironmentObject var router: Router
    var body: some View {
        AkScreen {
            TopBar(title: "Bilan de juin", subtitle: "Kouadio · 3ᵉ", onBack: { router.back() })
            ScrollView {
                VStack(alignment: .leading, spacing: 0) {
                    VStack(alignment: .leading, spacing: 0) {
                        HStack(alignment: .bottom) {
                            VStack(alignment: .leading, spacing: 0) {
                                Text("Moyenne générale").font(AkFont.regular(12)).foregroundColor(Ak.muted)
                                HStack(alignment: .bottom, spacing: 0) { Text("13,2").font(AkFont.schibstedBlack(30)).foregroundColor(Ak.ink); Text(" /20").font(AkFont.regular(13)).foregroundColor(Ak.faint).padding(.bottom, 3) }
                            }
                            Spacer()
                            HStack(spacing: 3) { Image(systemName: "arrow.up.right").font(.system(size: 11, weight: .bold)); Text("+1,4").font(AkFont.bold(13)) }.foregroundColor(Ak.green)
                                .padding(.horizontal, 11).padding(.vertical, 6).background(Ak.greenSoft).clipShape(RoundedRectangle(cornerRadius: 9))
                        }
                        HStack(alignment: .bottom, spacing: 9) { bar(0.56, "Mars", false); bar(0.59, "Avr", false); bar(0.64, "Mai", false); bar(0.84, "Juin", true) }.frame(height: 78).padding(.top, 16)
                    }.akCard(radius: 18, padding: 16)

                    HStack(spacing: 10) {
                        strength("Points forts", Ak.green, ["Calcul littéral", "Géométrie"], warn: false)
                        strength("À travailler", Ak.orange, ["Fonctions affines", "Probabilités"], warn: true)
                    }.padding(.top, 13)

                    VStack(alignment: .leading, spacing: 0) {
                        HStack(spacing: 9) {
                            Text("KN").font(AkFont.schibstedExtra(11)).foregroundColor(.white).frame(width: 30, height: 30).background(Ak.green).clipShape(RoundedRectangle(cornerRadius: 9))
                            Text("Mot du professeur").font(AkFont.bold(12.5)).foregroundColor(Ak.ink)
                        }
                        Text("« Kouadio progresse bien et gagne en confiance. On vise 14/20 au prochain bilan en consolidant les fonctions. »")
                            .font(AkFont.regular(12.5)).foregroundColor(Color(hex: 0x3F6B59)).lineSpacing(3).padding(.top, 8)
                    }.padding(14).frame(maxWidth: .infinity, alignment: .leading).background(Ak.greenSoft).clipShape(RoundedRectangle(cornerRadius: 16, style: .continuous)).padding(.top, 13)
                }.padding(.horizontal, 22).padding(.top, 14)
            }
            HStack(spacing: 0) {
                Image(systemName: "arrow.down.to.line").font(.system(size: 16)).foregroundColor(Ak.green)
                Text("  Télécharger le bilan (PDF)").font(AkFont.bold(14.5)).foregroundColor(Ak.green)
            }.frame(maxWidth: .infinity).padding(.vertical, 15).overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.green, lineWidth: 1))
            .padding(.horizontal, 22).padding(.vertical, 14).background(.white)
        }
    }
    func bar(_ f: CGFloat, _ l: String, _ active: Bool) -> some View {
        VStack(spacing: 6) {
            GeometryReader { geo in VStack { Spacer(minLength: 0); RoundedRectangle(cornerRadius: 6).fill(active ? Ak.green : Ak.greenSoft).frame(height: geo.size.height * f) } }
            Text(l).font(active ? AkFont.bold(9.5) : AkFont.regular(9.5)).foregroundColor(active ? Ak.green : Ak.faint)
        }.frame(maxWidth: .infinity)
    }
    func strength(_ title: String, _ color: Color, _ items: [String], warn: Bool) -> some View {
        VStack(alignment: .leading, spacing: 0) {
            Text(title).font(AkFont.bold(11.5)).foregroundColor(color).padding(.bottom, 7)
            ForEach(items, id: \.self) { i in
                HStack(spacing: 5) {
                    Image(systemName: warn ? "exclamationmark.circle.fill" : "checkmark.circle.fill").font(.system(size: 11)).foregroundColor(warn ? Ak.orange : Ak.online)
                    Text(i).font(AkFont.regular(12)).foregroundColor(Ak.ink)
                }.padding(.bottom, 4)
            }
        }.frame(maxWidth: .infinity, alignment: .leading).padding(12).background(.white).clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.border, lineWidth: 1))
    }
}

// MARK: - Écran 33, Reçu / facture
struct ReceiptScreen: View {
    @EnvironmentObject var router: Router
    var body: some View {
        AkScreen {
            TopBar(title: "Reçu de paiement", onBack: { router.back() })
            ScrollView {
                VStack(alignment: .leading, spacing: 0) {
                    VStack(alignment: .leading, spacing: 0) {
                        HStack {
                            HStack(spacing: 9) {
                                Image(systemName: "graduationcap.fill").font(.system(size: 18)).foregroundColor(.white).frame(width: 34, height: 34).background(Ak.green).clipShape(RoundedRectangle(cornerRadius: 10))
                                HStack(spacing: 0) { Text("Mon Prof Perso").font(AkFont.schibstedExtra(17)).foregroundColor(Ak.ink); Text(".").font(AkFont.schibstedExtra(17)).foregroundColor(Ak.orange) }
                            }
                            Spacer()
                            HStack(spacing: 5) { Image(systemName: "checkmark.circle.fill").font(.system(size: 12)); Text("Payé").font(AkFont.bold(11)) }.foregroundColor(Ak.green)
                                .padding(.horizontal, 10).padding(.vertical, 5).background(Ak.greenSoft).clipShape(RoundedRectangle(cornerRadius: 8))
                        }
                        Rectangle().fill(Ak.cardField).frame(height: 1).padding(.vertical, 16)
                        Text("REÇU N° AKW-2058").font(AkFont.bold(11.5)).foregroundColor(Ak.faint)
                        Text("Émis le 20 juin 2026").font(AkFont.regular(12.5)).foregroundColor(Ak.muted).padding(.top, 3)
                        HStack(alignment: .top) {
                            VStack(alignment: .leading, spacing: 0) {
                                Text("Cours de Maths · 3ᵉ").font(AkFont.semibold(13.5)).foregroundColor(Ak.ink)
                                Text("Koffi N'Guessan · 1h30").font(AkFont.regular(12)).foregroundColor(Ak.faint)
                            }
                            Spacer()
                            Text("6 000 F").font(AkFont.bold(13.5)).foregroundColor(Ak.ink)
                        }.padding(.top, 16)
                        HStack { Text("Frais de service").font(AkFont.regular(13)).foregroundColor(Ak.muted); Spacer(); Text("Offert").font(AkFont.semibold(13)).foregroundColor(Ak.green) }.padding(.top, 11)
                        Rectangle().fill(Ak.cardField).frame(height: 1).padding(.vertical, 14)
                        HStack { Text("Total payé").font(AkFont.bold(14)).foregroundColor(Ak.ink); Spacer(); Text("6 000 F").font(AkFont.schibstedExtra(20)).foregroundColor(Ak.green) }
                        HStack(spacing: 10) {
                            Image(systemName: "iphone").font(.system(size: 16)).foregroundColor(.white).frame(width: 32, height: 32).background(Ak.orangeMoney).clipShape(RoundedRectangle(cornerRadius: 9))
                            VStack(alignment: .leading, spacing: 0) { Text("Orange Money").font(AkFont.bold(12.5)).foregroundColor(Ak.ink); Text("Transaction OM240620.1432.A58").font(AkFont.regular(11)).foregroundColor(Ak.faint) }
                            Spacer()
                        }.padding(12).background(Ak.canvas).clipShape(RoundedRectangle(cornerRadius: 12)).padding(.top, 16)
                        HStack { Spacer()
                            Image(systemName: "qrcode").font(.system(size: 72)).foregroundColor(.white).frame(width: 96, height: 96).background(Ak.ink).clipShape(RoundedRectangle(cornerRadius: 12))
                            Spacer() }.padding(.top, 16)
                    }.akCard(radius: 20, padding: 20)
                }.padding(.horizontal, 22).padding(.top, 16)
            }
            HStack(spacing: 10) {
                ShareLink(item: "Reçu Mon Prof Perso N° AKW-2058, Cours de Maths · 3ᵉ avec Koffi N'Guessan, Total payé 6 000 F (Orange Money).") {
                    HStack(spacing: 0) { Image(systemName: "square.and.arrow.up").font(.system(size: 15)).foregroundColor(Ak.inkSoft); Text("  Partager").font(AkFont.bold(14)).foregroundColor(Ak.inkSoft) }
                        .frame(maxWidth: .infinity).padding(.vertical, 15).background(.white).clipShape(RoundedRectangle(cornerRadius: 14)).overlay(RoundedRectangle(cornerRadius: 14).stroke(Ak.border, lineWidth: 1))
                }
                HStack(spacing: 0) { Image(systemName: "arrow.down.to.line").font(.system(size: 15)).foregroundColor(.white); Text("  Télécharger (PDF)").font(AkFont.bold(14)).foregroundColor(.white) }
                    .frame(maxWidth: .infinity).padding(.vertical, 15).background(Ak.green).clipShape(RoundedRectangle(cornerRadius: 14))
                    .contentShape(Rectangle()).onTapGesture { router.back() }
            }.padding(.horizontal, 22).padding(.vertical, 14).background(.white)
        }
    }
}

File: android/app/src/main/java/ci/monprofperso/app/ui/screens/ManageScreens.kt
package ci.monprofperso.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Chat
import androidx.compose.material.icons.filled.*
import androidx.compose.foundation.clickable
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableIntStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import ci.monprofperso.app.nav.NavActions
import ci.monprofperso.app.nav.Routes
import ci.monprofperso.app.ui.components.*
import ci.monprofperso.app.ui.theme.AkColors
import ci.monprofperso.app.ui.theme.Hanken
import ci.monprofperso.app.ui.theme.Schibsted

/* ====================================================================== *
 * ÉCRAN 31, GÉRER LE COURS (ANNULER / REPROGRAMMER)
 * ====================================================================== */
@Composable
fun ManageCourseScreen(nav: NavActions) {
    var slot by remember { mutableIntStateOf(1) }
    AkScreen {
        TopBar("Détail du cours", onBack = { nav.back() })
        Column(Modifier.weight(1f).verticalScrollSafe().padding(horizontal = 22.dp).padding(top = 16.dp)) {
            Column(Modifier.fillMaxWidth().clip(RoundedCornerShape(18.dp)).background(AkColors.White).border(1.dp, AkColors.Border, RoundedCornerShape(18.dp)).padding(16.dp)) {
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                    Text("Maths · 3ᵉ", fontFamily = Schibsted, fontWeight = FontWeight.ExtraBold, fontSize = 16.sp, color = AkColors.Ink)
                    Row(Modifier.clip(RoundedCornerShape(8.dp)).background(AkColors.GreenSoft).padding(horizontal = 10.dp, vertical = 5.dp), verticalAlignment = Alignment.CenterVertically) {
                        Box(Modifier.size(6.dp).clip(CircleShape).background(Color(0xFF22A55D)))
                        Text("  Confirmé", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 11.sp, color = AkColors.Green)
                    }
                }
                Row(Modifier.padding(top = 13.dp), verticalAlignment = Alignment.CenterVertically) {
                    InitialsAvatar("KN", size = 42, radius = 12, fontSize = 15)
                    Spacer(Modifier.width(11.dp))
                    Column(Modifier.weight(1f)) {
                        Text("Koffi N'Guessan", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = AkColors.Ink)
                        Text("Professeur de Maths", fontFamily = Hanken, fontSize = 12.sp, color = AkColors.Muted)
                    }
                    IconSquare(Icons.AutoMirrored.Filled.Chat, bg = AkColors.CardField, tint = AkColors.Green, size = 38, onClick = { nav.go(ci.monprofperso.app.nav.Routes.Messaging) })
                }
                Box(Modifier.fillMaxWidth().padding(vertical = 13.dp).height(1.dp).background(AkColors.CardField))
                IconLine(Icons.Filled.DateRange, "Samedi 22 juin · 16h00 · 1h30")
                IconLine(Icons.Filled.Home, "À domicile · Cocody, Riviera 3")
                IconLine(Icons.Filled.AccountBalanceWallet, "6 000 F · payé via Orange Money", last = true)
            }
            Spacer(Modifier.height(14.dp))
            Row(Modifier.fillMaxWidth().clip(RoundedCornerShape(15.dp)).background(Color(0xFFFBF1E6)).padding(14.dp), verticalAlignment = Alignment.Top) {
                Icon(Icons.Filled.Info, null, tint = AkColors.Orange, modifier = Modifier.size(19.dp))
                Spacer(Modifier.width(10.dp))
                Column {
                    Text("Conditions d'annulation", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 13.sp, color = AkColors.Ink)
                    Text("Gratuit jusqu'à 24h avant le cours. Au-delà, 50% du montant est facturé au professeur pour son déplacement.",
                        fontFamily = Hanken, fontSize = 12.sp, lineHeight = 18.sp, color = Color(0xFF7A5733), modifier = Modifier.padding(top = 3.dp))
                }
            }
            Spacer(Modifier.height(14.dp))
            Text("Reprogrammer, nouveaux créneaux", fontFamily = Schibsted, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = AkColors.Ink)
            Spacer(Modifier.height(11.dp))
            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                listOf("Dim 23 · 14h", "Lun 24 · 17h", "Mar 25 · 16h").forEachIndexed { i, label ->
                    Chip(label, slot == i) { slot = i }
                }
            }
            Spacer(Modifier.height(16.dp))
        }
        Row(Modifier.fillMaxWidth().background(AkColors.White).padding(horizontal = 22.dp).padding(top = 14.dp, bottom = 24.dp), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
            Box(Modifier.weight(1f).clip(RoundedCornerShape(14.dp)).border(1.dp, Color(0xFFE0392B), RoundedCornerShape(14.dp)).clickable { nav.go(Routes.MyCourses) }.padding(vertical = 15.dp), contentAlignment = Alignment.Center) {
                Text("Annuler", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = Color(0xFFE0392B))
            }
            Box(Modifier.weight(1.4f).clip(RoundedCornerShape(14.dp)).background(AkColors.Green).clickable { nav.go(Routes.MyCourses) }.padding(vertical = 15.dp), contentAlignment = Alignment.Center) {
                Text("Confirmer le report", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = AkColors.White)
            }
        }
    }
}

@Composable
private fun IconLine(icon: ImageVector, text: String, last: Boolean = false) {
    Row(Modifier.padding(bottom = if (last) 0.dp else 8.dp), verticalAlignment = Alignment.CenterVertically) {
        Icon(icon, null, tint = AkColors.Green, modifier = Modifier.size(16.dp))
        Spacer(Modifier.width(9.dp))
        Text(text, fontFamily = Hanken, fontSize = 13.sp, color = Color(0xFF4A574F))
    }
}

/* ====================================================================== *
 * ÉCRAN 32, BILAN DE PROGRESSION DÉTAILLÉ
 * ====================================================================== */
@Composable
fun ProgressReportScreen(nav: NavActions) {
    AkScreen {
        TopBar("Bilan de juin", subtitle = "Kouadio · 3ᵉ", onBack = { nav.back() })
        Column(Modifier.weight(1f).verticalScrollSafe().padding(horizontal = 22.dp).padding(top = 14.dp)) {
            Column(Modifier.fillMaxWidth().clip(RoundedCornerShape(18.dp)).background(AkColors.White).border(1.dp, AkColors.Border, RoundedCornerShape(18.dp)).padding(16.dp)) {
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.Bottom) {
                    Column {
                        Text("Moyenne générale", fontFamily = Hanken, fontSize = 12.sp, color = AkColors.Muted)
                        Row(Modifier.padding(top = 2.dp), verticalAlignment = Alignment.Bottom) {
                            Text("13,2", fontFamily = Schibsted, fontWeight = FontWeight.Black, fontSize = 30.sp, color = AkColors.Ink)
                            Text(" /20", fontFamily = Hanken, fontSize = 13.sp, color = AkColors.Faint, modifier = Modifier.padding(bottom = 3.dp))
                        }
                    }
                    Row(Modifier.clip(RoundedCornerShape(9.dp)).background(AkColors.GreenSoft).padding(horizontal = 11.dp, vertical = 6.dp), verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Filled.TrendingUp, null, tint = AkColors.Green, modifier = Modifier.size(13.dp))
                        Text(" +1,4", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 13.sp, color = AkColors.Green)
                    }
                }
                Row(Modifier.fillMaxWidth().height(78.dp).padding(top = 16.dp), horizontalArrangement = Arrangement.spacedBy(9.dp), verticalAlignment = Alignment.Bottom) {
                    Bar2(0.56f, "Mars", false, Modifier.weight(1f))
                    Bar2(0.59f, "Avr", false, Modifier.weight(1f))
                    Bar2(0.64f, "Mai", false, Modifier.weight(1f))
                    Bar2(0.84f, "Juin", true, Modifier.weight(1f))
                }
            }
            Spacer(Modifier.height(13.dp))
            Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
                StrengthCard("Points forts", AkColors.Green, listOf("Calcul littéral", "Géométrie"), warn = false, Modifier.weight(1f))
                StrengthCard("À travailler", AkColors.Orange, listOf("Fonctions affines", "Probabilités"), warn = true, Modifier.weight(1f))
            }
            Spacer(Modifier.height(13.dp))
            Column(Modifier.fillMaxWidth().clip(RoundedCornerShape(16.dp)).background(AkColors.GreenSoft).padding(14.dp)) {
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Box(Modifier.size(30.dp).clip(RoundedCornerShape(9.dp)).background(AkColors.Green), contentAlignment = Alignment.Center) {
                        Text("KN", fontFamily = Schibsted, fontWeight = FontWeight.ExtraBold, fontSize = 11.sp, color = AkColors.White)
                    }
                    Spacer(Modifier.width(9.dp))
                    Text("Mot du professeur", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 12.5.sp, color = AkColors.Ink)
                }
                Text("« Kouadio progresse bien et gagne en confiance. On vise 14/20 au prochain bilan en consolidant les fonctions. »",
                    fontFamily = Hanken, fontSize = 12.5.sp, lineHeight = 19.sp, color = Color(0xFF3F6B59), modifier = Modifier.padding(top = 8.dp))
            }
            Spacer(Modifier.height(16.dp))
        }
        Box(Modifier.fillMaxWidth().background(AkColors.White).padding(horizontal = 22.dp).padding(top = 14.dp, bottom = 24.dp)) {
            Row(Modifier.fillMaxWidth().clip(RoundedCornerShape(14.dp)).border(1.dp, AkColors.Green, RoundedCornerShape(14.dp)).padding(vertical = 15.dp), horizontalArrangement = Arrangement.Center, verticalAlignment = Alignment.CenterVertically) {
                Icon(Icons.Filled.Download, null, tint = AkColors.Green, modifier = Modifier.size(18.dp))
                Text("  Télécharger le bilan (PDF)", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.5.sp, color = AkColors.Green)
            }
        }
    }
}

@Composable
private fun Bar2(fraction: Float, label: String, active: Boolean, modifier: Modifier = Modifier) {
    Column(modifier.fillMaxHeight(), horizontalAlignment = Alignment.CenterHorizontally, verticalArrangement = Arrangement.Bottom) {
        Box(Modifier.fillMaxWidth().fillMaxHeight(fraction).clip(RoundedCornerShape(topStart = 6.dp, topEnd = 6.dp)).background(if (active) AkColors.Green else AkColors.GreenSoft))
        Spacer(Modifier.height(6.dp))
        Text(label, fontFamily = Hanken, fontWeight = if (active) FontWeight.Bold else FontWeight.Normal, fontSize = 9.5.sp, color = if (active) AkColors.Green else AkColors.Faint)
    }
}

@Composable
private fun StrengthCard(title: String, color: Color, items: List<String>, warn: Boolean, modifier: Modifier = Modifier) {
    Column(modifier.clip(RoundedCornerShape(14.dp)).background(AkColors.White).border(1.dp, AkColors.Border, RoundedCornerShape(14.dp)).padding(12.dp)) {
        Text(title, fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 11.5.sp, color = color, modifier = Modifier.padding(bottom = 7.dp))
        items.forEach {
            Row(Modifier.padding(bottom = 4.dp), verticalAlignment = Alignment.CenterVertically) {
                Icon(if (warn) Icons.Filled.Error else Icons.Filled.CheckCircle, null, tint = if (warn) AkColors.Orange else Color(0xFF22A55D), modifier = Modifier.size(13.dp))
                Spacer(Modifier.width(5.dp))
                Text(it, fontFamily = Hanken, fontSize = 12.sp, color = AkColors.Ink)
            }
        }
    }
}

/* ====================================================================== *
 * ÉCRAN 33, REÇU / FACTURE
 * ====================================================================== */
@Composable
fun ReceiptScreen(nav: NavActions) {
    val context = androidx.compose.ui.platform.LocalContext.current
    val receiptText = "Reçu Mon Prof Perso N° AKW-2058, Cours de Maths · 3ᵉ avec Koffi N'Guessan, Total payé 6 000 F (Orange Money)."
    AkScreen {
        TopBar("Reçu de paiement", onBack = { nav.back() })
        Column(Modifier.weight(1f).verticalScrollSafe().padding(horizontal = 22.dp).padding(top = 16.dp)) {
            Column(Modifier.fillMaxWidth().clip(RoundedCornerShape(20.dp)).background(AkColors.White).border(1.dp, AkColors.Border, RoundedCornerShape(20.dp)).padding(20.dp)) {
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                    Row(verticalAlignment = Alignment.CenterVertically) {
                        Box(Modifier.size(34.dp).clip(RoundedCornerShape(10.dp)).background(AkColors.Green), contentAlignment = Alignment.Center) {
                            Icon(Icons.Filled.School, null, tint = AkColors.White, modifier = Modifier.size(18.dp))
                        }
                        Spacer(Modifier.width(9.dp))
                        Row {
                            Text("Mon Prof Perso", fontFamily = Schibsted, fontWeight = FontWeight.ExtraBold, fontSize = 17.sp, color = AkColors.Ink)
                            Text(".", fontFamily = Schibsted, fontWeight = FontWeight.ExtraBold, fontSize = 17.sp, color = AkColors.Orange)
                        }
                    }
                    Row(Modifier.clip(RoundedCornerShape(8.dp)).background(AkColors.GreenSoft).padding(horizontal = 10.dp, vertical = 5.dp), verticalAlignment = Alignment.CenterVertically) {
                        Icon(Icons.Filled.CheckCircle, null, tint = AkColors.Green, modifier = Modifier.size(13.dp))
                        Text("  Payé", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 11.sp, color = AkColors.Green)
                    }
                }
                Box(Modifier.fillMaxWidth().padding(vertical = 16.dp).height(1.dp).background(AkColors.CardField))
                Text("REÇU N° AKW-2058", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 11.5.sp, color = AkColors.Faint)
                Text("Émis le 20 juin 2026", fontFamily = Hanken, fontSize = 12.5.sp, color = AkColors.Muted, modifier = Modifier.padding(top = 3.dp))
                Spacer(Modifier.height(16.dp))
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.Top) {
                    Column {
                        Text("Cours de Maths · 3ᵉ", fontFamily = Hanken, fontWeight = FontWeight.SemiBold, fontSize = 13.5.sp, color = AkColors.Ink)
                        Text("Koffi N'Guessan · 1h30", fontFamily = Hanken, fontSize = 12.sp, color = AkColors.Faint)
                    }
                    Text("6 000 F", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 13.5.sp, color = AkColors.Ink)
                }
                Row(Modifier.fillMaxWidth().padding(top = 11.dp), horizontalArrangement = Arrangement.SpaceBetween) {
                    Text("Frais de service", fontFamily = Hanken, fontSize = 13.sp, color = AkColors.Muted)
                    Text("Offert", fontFamily = Hanken, fontWeight = FontWeight.SemiBold, fontSize = 13.sp, color = AkColors.Green)
                }
                Box(Modifier.fillMaxWidth().padding(vertical = 14.dp).height(1.dp).background(AkColors.CardField))
                Row(Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween, verticalAlignment = Alignment.CenterVertically) {
                    Text("Total payé", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = AkColors.Ink)
                    Text("6 000 F", fontFamily = Schibsted, fontWeight = FontWeight.ExtraBold, fontSize = 20.sp, color = AkColors.Green)
                }
                Row(Modifier.fillMaxWidth().padding(top = 16.dp).clip(RoundedCornerShape(12.dp)).background(AkColors.Canvas).padding(12.dp), verticalAlignment = Alignment.CenterVertically) {
                    Box(Modifier.size(32.dp).clip(RoundedCornerShape(9.dp)).background(Color(0xFFF2761A)), contentAlignment = Alignment.Center) {
                        Icon(Icons.Filled.PhoneAndroid, null, tint = AkColors.White, modifier = Modifier.size(16.dp))
                    }
                    Spacer(Modifier.width(10.dp))
                    Column {
                        Text("Orange Money", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 12.5.sp, color = AkColors.Ink)
                        Text("Transaction OM240620.1432.A58", fontFamily = Hanken, fontSize = 11.sp, color = AkColors.Faint)
                    }
                }
                Box(Modifier.fillMaxWidth().padding(top = 16.dp), contentAlignment = Alignment.Center) {
                    Box(Modifier.size(96.dp).clip(RoundedCornerShape(12.dp)).background(AkColors.Ink), contentAlignment = Alignment.Center) {
                        Icon(Icons.Filled.QrCode2, null, tint = AkColors.White, modifier = Modifier.size(72.dp))
                    }
                }
            }
            Spacer(Modifier.height(16.dp))
        }
        Row(Modifier.fillMaxWidth().background(AkColors.White).padding(horizontal = 22.dp).padding(top = 14.dp, bottom = 24.dp), horizontalArrangement = Arrangement.spacedBy(10.dp)) {
            Row(Modifier.weight(1f).clip(RoundedCornerShape(14.dp)).background(AkColors.White).border(1.dp, AkColors.Border, RoundedCornerShape(14.dp)).clickable { shareText(context, receiptText) }.padding(vertical = 15.dp), horizontalArrangement = Arrangement.Center, verticalAlignment = Alignment.CenterVertically) {
                Icon(Icons.Filled.Share, null, tint = AkColors.InkSoft, modifier = Modifier.size(17.dp))
                Text("  Partager", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = AkColors.InkSoft)
            }
            Row(Modifier.weight(1.3f).clip(RoundedCornerShape(14.dp)).background(AkColors.Green).clickable { nav.back() }.padding(vertical = 15.dp), horizontalArrangement = Arrangement.Center, verticalAlignment = Alignment.CenterVertically) {
                Icon(Icons.Filled.Download, null, tint = AkColors.White, modifier = Modifier.size(17.dp))
                Text("  Télécharger (PDF)", fontFamily = Hanken, fontWeight = FontWeight.Bold, fontSize = 14.sp, color = AkColors.White)
            }
        }
    }
}
